### DISCUSSION TITLE
How to store SPLUNK_ACCESS_TOKEN use by OTEL library in spring boot application

### DISCUSSION THREAD

**Discussion by furkanali0325** [signalfx/splunk-otel-java#846] [Q&A]
Hi Team,

I have a spring boot application, in which I have imported the Splunk OTEL library(Splunk open telemetry jar file [https://github.com/signalfx/splunk-otel-java][1]) for tracing and matrices. 
  So a Splunk authentication token i.e Splunk access token is needed to export data directly to Splunk Observability Cloud. Currently, I have exposed the Splunk access token in Kubernetes.yaml file as an env variable.
But I don't want to expose this token by using Kubernetes.yaml or using a properties file.
So I just want to know, is there any other way to expose or store the splunk_acces_token apart from Kubernetes.yaml or properties file?

For example: there is a flag variable "OTEL_JAVAAGENT_ENABLED" used by the Splunk OTEL library and I have stored this value to "true" in kubernetes.yaml file. So by this config OTEL jar is able to access it.

**Comment by mateuszrzeszutek** [comment]
Hey @furkanali0325 ,
I'm no expert on kubernetes, but I'm pretty sure that it must have some secret vault functionality for these kind of cases (every CI/CD tool does). Try https://kubernetes.io/docs/concepts/configuration/secret/, maybe that'll point you in the right direction.
